File: wi-seong-cheol/BOJ/0x03 Array/3273.swift
//
//  3273.swift
//  wi-seong-cheol
//
//  Created by wi_seong on 2022/12/14.
//

/*
 [Input]
 9
 5 12 7 10 9 1 2 3 11
 13
 [Output]
 3
 */
// 시간복잡도: O(n)
// 최악시간: 1000000
// 난이도: Silver 3
// Timer: 5m
// Url: https://www.acmicpc.net/problem/3273

import Foundation

let n: Int = Int(readLine()!)!
let arr: [Int] = readLine()!.split(separator: " ").map { Int(String($0))! }
let x: Int = Int(readLine()!)!

var check: [Int] = [Int](repeating: 0, count: 2000001)

for i in 0..<n {
    check[arr[i]] = 1
}

var ans: Int = 0
for i in 0..<(x + 1)/2 {
    if check[i] != 0 && check[x - i] != 0 {
        ans += 1
    }
}

print(ans)

File: hall_of_fame/wi-seong-cheol/BOJ/0x03 Array/3273.cpp
//
//  3273.cpp
//  main
//
//  Created by wi_seong on 2022/11/06.
//

/*
 [Input]
 9
 5 12 7 10 9 1 2 3 11
 13
 [Output]
 3
 */
// 시간복잡도: O(n)
// 최악시간: 1000000
// 난이도: Silver 3
// Timer: 5m
// Url: https://www.acmicpc.net/problem/3273

#include <iostream>

using namespace std;
int n, x;
int arr[2000001];
int ans;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    cin >> n;
    for(int i = 0; i < n; i++) {
        int a; cin >> a;
        arr[a] = 1;
    }
    cin >> x;
    for(int i = 1; i < (x + 1)/2; i++)
        if(arr[i] && arr[x - i]) ans++;
    cout << ans;
    
    return 0;
}
